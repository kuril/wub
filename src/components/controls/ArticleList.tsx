import * as React from "react";
import { connect } from 'react-redux';
import { Article, RootState } from "types/wub";

type ArticlesListStateProps = { list: Article[], isLoading: boolean, loadedAt: Date | null }
type ArticlesListProps = ArticlesListStateProps

const ArticlesList: React.SFC<ArticlesListProps> = ({ list, isLoading, loadedAt }) =>
    (<div className="page-container">
        {list.length < 1 && !isLoading && <small>No articles</small>}
        {isLoading && <small>Loading articles</small>}
        {list.length > 0 && !isLoading && <div className="box">{
            list.map(
                (article, i) => (<article key={i} className="media">
                    <div className="media-content">
                        <strong>{article.title}</strong>
                        <small>{article.author}</small>
                        <p>{article.body}</p>
                    </div>
                </article>))
        }</div>}
        {loadedAt && !isLoading && <small>Last updated at: {loadedAt.toLocaleString()}</small>}
    </div>)

const mapStateToProps = (state: RootState) => {
    const { list, isLoading, loadedAt } = state.article;
    return {
        list: list.concat(),
        isLoading,
        loadedAt: loadedAt != null ? new Date(loadedAt) : null,
    };
};

export const ArticleListConnected = connect<ArticlesListStateProps>(mapStateToProps)(ArticlesList)