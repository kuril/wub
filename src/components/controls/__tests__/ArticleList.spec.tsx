import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { ArticlesList } from "../ArticleList";
import { Article } from "types/wub";
import { MemoryRouter } from "react-router-dom"

const articles: Article[] = [
    { id: 1, title: "article1", body: "article1 body", author: "test author"},
    { id: 2, title: "article2", body: "article2 body", author: "test author"},
    { id: 3, title: "article3", body: "article2 body", author: "test author"},
]

describe("ActionList component", () => {
    it("renders correctly", () => {
        expect(TestRenderer.create(<MemoryRouter>
            <ArticlesList isLoading={false} loadedAt={null} list={articles} />
        </MemoryRouter>           
        )).toMatchSnapshot()
    })
})