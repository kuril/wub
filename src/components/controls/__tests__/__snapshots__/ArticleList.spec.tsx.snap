// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionList component renders correctly 1`] = `
<div
  className="page-container"
>
  <div
    className="box"
  >
    <article
      className="media"
    >
      <div
        className="media-content"
      >
        <strong>
          article1
        </strong>
        <small>
          test author
        </small>
        <p>
          article1 body
        </p>
        <a
          href="/comments/1"
          onClick={[Function]}
        >
          <small>
            comments
          </small>
        </a>
      </div>
    </article>
    <article
      className="media"
    >
      <div
        className="media-content"
      >
        <strong>
          article2
        </strong>
        <small>
          test author
        </small>
        <p>
          article2 body
        </p>
        <a
          href="/comments/2"
          onClick={[Function]}
        >
          <small>
            comments
          </small>
        </a>
      </div>
    </article>
    <article
      className="media"
    >
      <div
        className="media-content"
      >
        <strong>
          article3
        </strong>
        <small>
          test author
        </small>
        <p>
          article2 body
        </p>
        <a
          href="/comments/3"
          onClick={[Function]}
        >
          <small>
            comments
          </small>
        </a>
      </div>
    </article>
  </div>
</div>
`;
